### PR TITLE
[#1293]feat(rust): Add total_read_data metric

### PR DESCRIPTION
**Pull request by wenlongbrother**
### What changes were proposed in this pull request?

Add total_received_data metric

(Please outline the changes and how this PR fixes the issue.)

### Why are the changes needed?
Currently, the total_received_data metric has been introduced, the total_read_data is also necessary

### Does this PR introduce _any_ user-facing change?

(Please list the user-facing changes introduced by your change, including
  1. Change in user-facing APIs.
  2. Addition or removal of property keys.)

No.

### How was this patch tested?

(Please test your changes, and provide instructions on how to test it:
  1. If you add a feature or fix a bug, add a test to cover your changes. 
  2. If you fix a flaky test, repeat it for many times to prove it works.)

### CODE REVIEW

File: rust/experimental/server/src/store/hybrid.rs [hunk 1]
```diff
@@ -432,10 +428,14 @@ impl Store for HybridStore {
 
     async fn get(&self, ctx: ReadingViewContext) -> Result<ResponseData, WorkerError> {
         match ctx.reading_options {
-            ReadingOptions::MEMORY_LAST_BLOCK_ID_AND_MAX_SIZE(_, _) => {
+            ReadingOptions::MEMORY_LAST_BLOCK_ID_AND_MAX_SIZE(_, length) => {
```
Comment (zuston) [inline rust/experimental/server/src/store/hybrid.rs:431]: We'd better to get the read size in app.rs

File: rust/experimental/server/src/app.rs [hunk 2]
```diff
@@ -183,6 +183,13 @@ impl App {
...
 
     pub async fn select(&self, ctx: ReadingViewContext) -> Result<ResponseData, WorkerError> {
+        match ctx.reading_options {
+            ReadingOptions::MEMORY_LAST_BLOCK_ID_AND_MAX_SIZE(_, length)
+            | ReadingOptions::FILE_OFFSET_AND_LEN(_, length) => {
+                TOTAL_READ_DATA.inc_by(length as u64);
+            }
+        }
+
         self.store.get(ctx).await
```
Comment (zuston) [inline rust/experimental/server/src/app.rs:193]: Why not extract the responseData to get the read size? The current change is not accurate, especially the reading is slow, and then the timeserial metrics will be incorrect.

**Review by zuston — COMMENTED** [review]
Overall the change looks good. left minor comments.

File: rust/experimental/server/src/app.rs (hunk 2 above)
Comment (wenlongbrother) [reply rust/experimental/server/src/app.rs:193]: make sense

**Review by zuston — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**